testing_and_setup/compass/ocean/global_ocean/scripts/create_E3SM_coupling_files.py

```diff
@@ -307,195 +334,158 @@ def mapping_analysis(config, mesh_name, date_string, ice_shelf_cavities):
             run_command(args)
 
             # Atmosphere to ocean
+            mapping_file = 'map_{}_TO_{}.mask_{}.{}.nc'.format(
+                atm_scrip_tag, mesh_name, short, date_string)
             args = ['mpirun', '-n', nprocs, 'ESMF_RegridWeightGen',
                     '--method', method,
                     '--source', atm_scrip_file,
                     '--destination', ocn_scrip_file,
-                    '--weight', 'map_' + atm_scrip_tag + '_TO_' + mesh_name + nomaskStr + '_'
-                       + short + '.' + date_string + '.nc',
+                    '--weight', mapping_file,
                     '--ignore_unmapped']
             run_command(args)
-
-    except OSError:
-        print('mapping must be run on one compute node')
-
-    if ice_shelf_cavities:
-        print("\n Mapping files with masks for ice shelf cavities")
-        # make links
-        ocn_scrip_file = 'ocean.' + mesh_name + '.mask.scrip.' + date_string + '.nc'
-        make_link('../scrip/' + ocn_scrip_file, ocn_scrip_file)
-
-        # execute commands
-        try:
-            for method, short in [['conserve', 'aave'], ['bilinear', 'blin'], ['patch', 'patc']]:
-
-                # Ocean to atmosphere
-                args = ['mpirun', '-n', nprocs, 'ESMF_RegridWeightGen',
-                        '--method', method,
-                        '--source', ocn_scrip_file,
-                        '--destination', atm_scrip_file,
-                        '--weight', 'map_' + mesh_name + '.mask_TO_' + atm_scrip_tag + '_'
-                           + short + '.' + date_string + '.nc',
-                        '--ignore_unmapped']
-                run_command(args)
-
-                # Atmosphere to ocean
-                args = ['mpirun', '-n', nprocs, 'ESMF_RegridWeightGen',
-                        '--method', method,
-                        '--source', atm_scrip_file,
-                        '--destination', ocn_scrip_file,
-                        '--weight', 'map_' + atm_scrip_tag + '_TO_' + mesh_name + '.mask_'
-                           + short + '.' + date_string + '.nc',
-                        '--ignore_unmapped']
-                run_command(args)
-
-        except OSError:
-            print('mapping_CORE_Gcase must be run on one compute node')
 # }}}
 
 
-def domain_CORE_Gcase(config, mesh_name, date_string, ice_shelf_cavities):
-# {{{
-    # obtain configuration settings
-    domain_exe = config.get('main', 'domain_exe')
-    atm_scrip_tag = config.get('mapping_CORE_Gcase', 'atm_scrip_tag')
-    if ice_shelf_cavities:
-        nomaskStr='.nomask'
-    else:
-        nomaskStr=''
-
-    # make links
-    make_link(domain_exe, 'domain_exe')
-    mapping_file = 'map_' + mesh_name + nomaskStr + '_TO_' + \
-        atm_scrip_tag + '_aave.' + date_string + '.nc'
-    make_link('../mapping_CORE_Gcase/' + mapping_file, mapping_file)
+def domain_CORE_Gcase(config):  # {{{
+    make_domain_files(config, 'CORE_Gcase')
+# }}}
```

Comment:
There are 3 essentially identical functions with a single string that differs between them.  It makes a lot more sense to have one function than 3 copies.